src/pages/workspaces/workspace/workflows/WorkflowView.js

```diff
@@ -325,6 +325,8 @@ const WorkflowView = _.flow(
   resetSelectionModel(value, selectedEntities = {}, entityMetadata = this.state.entityMetadata) {
     const { workflowName } = this.props
 
+    // If the default for non-set types changes from `processAllAsSet` then the calculation of `noLaunchReason` in `renderSummary` needs to be updated accordingly.
+    // Currently, `renderSummary` assumes that it is not possible to have nothing selected for non-set types.
```

Comment:
`chooseSets` is now somewhat redundant with `chooseRows`, but I've left it in order to keep this PR relatively small and because it still reflects the current user experience. @panentheos and I discussed this and concluded that there are 2 (potentially separate) efforts we could undertake:

1. Refactor the code for simplicity
2. Rework the UI for simplicity and clarity

As part of understanding the changes required for this PR, I audited all occurrences of `chooseSet`. While doing so, I concluded that the conditions in this reset and the conditions the logic for validating the selection model below are linked. Therefore, I've added comments in both places to help future refactor/rework efforts.